### DISCUSSION TITLE
Handling non-structured data

### DISCUSSION THREAD

**Discussion by zorgoz** [JasperFx/marten#2458] [Q&A]
I have a project in sight, where a document database is the best approach, and I absolutely have Marten in mind. However, I need to handle a mixture of strongly typed and dynamic data. MongoDB has such a feature: https://mongodb.github.io/mongo-csharp-driver/2.0/examples/mixing_static_and_dynamic/ . Does Marten also provide something similar - or is a simple IDictionary<string, object> and a proper JsonConverter the way?

**Comment by gfoidl** [comment]
Does [Storing & reading back non-uniform JSON documents via dynamic](https://martendb.io/scenarios/dynamic-data.html) help you?

**Comment by zorgoz** [comment]
> Does [Storing & reading back non-uniform JSON documents via dynamic](https://martendb.io/scenarios/dynamic-data.html) help you?

Yes, most likely, thank you.

**Comment by jeremydmiller** [comment]
I'm moving this to a discussion folks. And thank you to @gfoidl for being so active on these issues and helping other folks!
